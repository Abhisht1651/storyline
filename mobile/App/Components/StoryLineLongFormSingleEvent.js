// @flow
import React from 'react'
import PropTypes from 'prop-types'
import { View, Text, Image, ImageBackground, TouchableOpacity, ImageProps } from 'react-native'

// Styles
import styles from './Styles/StoryLineLongFormSingleEventStyle'
import images from '../Themes/Images'

type Props = {
  reach: number,
  heading: string,
  storyTitle: string,
  previewText: string,
  updatedTime: string,
  comments: number,
  onPressShare: PropTypes.func,
  onPressReact: PropTypes.func,
  onPressRaven: PropTypes.func,
  onPressComments: PropTypes.func,
  storyImage: ImageProps
}

const storyLineLongFormSingleEvent = (props: Props) => {
  const {
    reach,
    storyTitle,
    updatedTime,
    heading,
    previewText,
    comments,
    onPressReact,
    onPressRaven,
    onPressShare,
    onPressComments,
    storyImage
  } = props
  const {
    iconStoryLineInactive,
    iconStoryLineTrending,
    iconStoryLineReach,
    iconStoryLineRelatedEntity1,
    iconStoryLineRelatedEntity2,
    iconStoryLineRelatedEntity3,
    iconStoryLineRelatedEntity4,
    voterIcon1,
    voterIcon2,
    voterIcon3,
    shareIcon,
    reactIcon,
    ravenIcon,
    sourceIcon
  } = images

  const iconRelatedEntity = [
    iconStoryLineRelatedEntity1,
    iconStoryLineRelatedEntity2,
    iconStoryLineRelatedEntity3,
    iconStoryLineRelatedEntity4
  ]

  const voterIcons = [voterIcon1, voterIcon2, voterIcon3]

  const commentIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.commentsIcon} />
  ))

  const relatedEntityIconContainer = iconRelatedEntity.map((icon, index) => (
    <Image source={icon} key={index} style={styles.relatedEntityIcon} />
  ))

  return (
    <View style={styles.container}>
      <View style={styles.headerContainer}>
        <Text style={styles.headingText}>{heading}</Text>
        <View style={styles.dot} />
        <Image source={iconStoryLineInactive} style={styles.storyLineIconHeading} />
        <Text style={styles.headingText}>Inactive</Text>
        <View style={styles.dot} />
        <Image source={iconStoryLineTrending} style={styles.storyLineIconHeading} />
        <Text style={styles.headingText}>Trending</Text>
      </View>
      <View style={styles.reachContainer}>
        <Image source={iconStoryLineReach} style={styles.reachIcon} />
        <Text style={styles.reachText}>{reach}K Reached</Text>
      </View>
      <Text style={styles.newsTitleText}>{storyTitle}</Text>
      <Text style={styles.updateTimeText}>Updated {updatedTime} ago</Text>
      <View style={styles.previewContainer}>
        <ImageBackground source={storyImage} style={styles.image} imageStyle={styles.imageBorder}>
          <Image source={sourceIcon} style={styles.sourceImage} />
        </ImageBackground>
        <Text style={styles.previewText}>{previewText}</Text>
        <View style={styles.relatedEntityContainer}>
          <Text style={styles.EntityRelatedText}>Related entities</Text>
          <View style={styles.relatedEntityIconContainer}>{relatedEntityIconContainer}</View>
        </View>
        <View style={styles.divider} />
        <TouchableOpacity style={styles.commentsContainer} onPress={onPressComments}>
          {commentIconsElement}
          <Text style={styles.commentsText}>{comments}K comments</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.actionIconsContainer}>
        <TouchableOpacity onPress={onPressShare}>
          <Image source={shareIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressRaven}>
          <Image source={ravenIcon} style={styles.actionIcon} />
        </TouchableOpacity>
        <TouchableOpacity onPress={onPressReact}>
          <Image source={reactIcon} style={styles.actionIcon} />
        </TouchableOpacity>
      </View>
    </View>
  )
}

storyLineLongFormSingleEvent.defaultProps = {
  reach: 45,
  heading: 'World',
  storyTitle: 'UK exit from the European Union',
  previewText: 'Corbyn wins Labor conference Brexit vote in the latest polls',
  updatedTime: '2m',
  comments: 34,
  onPressReact: () => {},
  onPressRaven: () => {},
  onPressShare: () => {},
  onPressComments: () => {},
  storyImage: images.storyImage
}

export default storyLineLongFormSingleEvent
