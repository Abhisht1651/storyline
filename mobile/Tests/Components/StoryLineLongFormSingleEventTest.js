// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import StoryLineLongFormSingleEvent from '../../App/Components/StoryLineLongFormSingleEvent'

describe('StoryLineLongFormSingleEvent', () => {
  it('should render the component itself correctly', () => {
    const wrapper = shallow(<StoryLineLongFormSingleEvent />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<StoryLineLongFormSingleEvent />)
    expect(wrapper.find('Text')).toHaveLength(9)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<StoryLineLongFormSingleEvent />)
    expect(wrapper.find('Image')).toHaveLength(14)
  })
})
