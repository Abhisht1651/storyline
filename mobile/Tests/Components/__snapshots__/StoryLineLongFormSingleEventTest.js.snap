// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryLineLongFormSingleEvent should render the component itself correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E6ECED",
      "borderRadius": 30,
      "borderStyle": "solid",
      "borderWidth": 1,
      "minHeight": 570,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 50,
        "opacity": 1,
        "width": "85.23%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 28,
          "letterSpacing": -0.08,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      World
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#1F1F60",
          "borderRadius": 2,
          "height": 4,
          "marginHorizontal": 10,
          "width": 4,
        }
      }
    />
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 34,
          "marginRight": 10,
          "width": 34,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 28,
          "letterSpacing": -0.08,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Inactive
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#1F1F60",
          "borderRadius": 2,
          "height": 4,
          "marginHorizontal": 10,
          "width": 4,
        }
      }
    />
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 34,
          "marginRight": 10,
          "width": 34,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 28,
          "letterSpacing": -0.08,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Trending
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 22,
        "width": "85.23%",
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 26,
          "marginRight": 4,
          "opacity": 1,
          "width": 26,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8F8FAF",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 24,
          "letterSpacing": -0.07,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      45
      K Reached
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Medium",
        "fontSize": 32,
        "letterSpacing": -0.11,
        "marginTop": 48,
        "opacity": 1,
        "textAlign": "left",
        "width": "85.23%",
      }
    }
  >
    UK exit from the European Union
  </Text>
  <Text
    style={
      Object {
        "color": "#8F8FAF",
        "fontFamily": "BananaGrotesk-Regular",
        "fontSize": 24,
        "letterSpacing": -0.07,
        "marginTop": 5,
        "opacity": 1,
        "textAlign": "left",
        "width": "85.23%",
      }
    }
  >
    Updated 
    2m
     ago
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#E6ECED",
        "borderRadius": 30,
        "borderWidth": 0.5,
        "marginTop": 34,
        "opacity": 1,
        "width": "85.23%",
      }
    }
  >
    <ImageBackground
      imageStyle={
        Object {
          "borderTopLeftRadius": 30,
          "borderTopRightRadius": 30,
        }
      }
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "alignItems": "flex-end",
          "height": 316.5,
          "justifyContent": "flex-end",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": "15%",
            "width": "20%",
          }
        }
      />
    </ImageBackground>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 36,
          "letterSpacing": -0.12,
          "marginTop": "8%",
          "opacity": 1,
          "textAlign": "left",
          "width": "86.28%",
        }
      }
    >
      Corbyn wins Labor conference Brexit vote in the latest polls
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": "3%",
          "width": "86.28%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Medium",
            "fontSize": 24,
            "letterSpacing": -0.07,
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
        Related entities
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginLeft": "5%",
          }
        }
      >
        <Image
          key="0"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 8,
              "height": 37.5,
              "marginHorizontal": "3%",
              "width": 37.5,
            }
          }
        />
        <Image
          key="1"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 8,
              "height": 37.5,
              "marginHorizontal": "3%",
              "width": 37.5,
            }
          }
        />
        <Image
          key="2"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 8,
              "height": 37.5,
              "marginHorizontal": "3%",
              "width": 37.5,
            }
          }
        />
        <Image
          key="3"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 8,
              "height": 37.5,
              "marginHorizontal": "3%",
              "width": 37.5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E6ECED",
          "borderBottomWidth": 1,
          "marginTop": "4%",
          "width": "100%",
        }
      }
    />
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 10,
          "marginVertical": "6%",
          "width": "86.28%",
        }
      }
    >
      <Image
        key="0"
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 40,
            "marginLeft": -10,
            "opacity": 1,
            "width": 40,
          }
        }
      />
      <Image
        key="1"
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 40,
            "marginLeft": -10,
            "opacity": 1,
            "width": 40,
          }
        }
      />
      <Image
        key="2"
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "height": 40,
            "marginLeft": -10,
            "opacity": 1,
            "width": 40,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
            "letterSpacing": 0,
            "marginLeft": "2%",
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
        34
        K comments
      </Text>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 40,
        "width": "82.857%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
  </View>
</View>
`;
